corehq/ex-submodules/pillowtop/pillow/interface.py

```diff
@@ -99,6 +99,14 @@ def run(self):
         pillow_logging.info("Starting pillow %s" % self.__class__)
         self.process_changes(since=self.get_last_checkpoint_sequence(), forever=True)
 
+    def _update_checkpoint(self, change, context):
+        if change:
```

Comment:
if you make this `if change and context:` then you don't need to pass in the runtime context in the unit tests.

Reply:
https://github.com/dimagi/commcare-hq/tree/4e680f7dfda5623e9e287725b85b45aecd528c03